server/message_cache.go

```diff
@@ -700,6 +717,12 @@ func readMessage(rows *sql.Rows) (*message, error) {
 			return nil, err
 		}
 	}
+	var extras map[string]string
```

Comment:
What if, instead of a `map[string]string`, you used a `map[string]json.RawMessage`?

That way, you should be able to use any arbitrary JSON, instead of just strings.

https://play.golang.com/p/AZ2XzbdqnXk